src/base-vs/parameterResolver.ts

```diff
@@ -0,0 +1,36 @@
+import { Parameter, ParameterValues } from "../api";
+
+export interface ParameterResolver {
+    resolve: (parameterDefinition: Parameter) => string;
+    resolveOptional: (parameterDefinition: Parameter) => string | undefined;
+}
+
+function buildValuesMap(values: ParameterValues): Map<string, string> {
+    const valuesMap = new Map<string, string>()
+    for (const value of values.values) {
+        if (valuesMap.has(value.name)) {
+            throw new Error(`Two values '${value.value}' and '${valuesMap.get(value.name)}' found for parameter ${value.name}`)
+        }
+        valuesMap.set(value.name, value.value)
+    }
+    return valuesMap
+}
+
+export function createParameterResolver(paramValues: ParameterValues): ParameterResolver {
```

Comment:
Proposal: Rename to `ParameterAccessor` and methods `getOptional()` and `get()`.